.github/workflows/stale.yml

```diff
@@ -0,0 +1,105 @@
+name: Stale and close inactive issues/PRs
+
+on:
+  schedule:
+    - cron: "30 1 * * *"  # each day at 1:30AM
+  workflow_dispatch:      # or manually
+
+permissions:
+  issues: read
+  pull-requests: read
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.head_ref || github.ref || github.run_id }}'
+  cancel-in-progress: true
+
+jobs:
+
+  process-issues:
+    name: Process Issues
+
+    permissions:
+      issues: write
```

Comment:
Since it is an automated workflow, I don't think there should be any vulnerability issues.

To make the code simpler you could use [`probot/stale`](https://github.com/probot/stale), the bot could serve in other occasions (see https://github.com/probot/reminders, https://probot.github.io/apps/...).

Can you test this workflow? Maybe temporarily reducing the days to 1/2.

Reply:
> Since it is an automated workflow, I don't think there should be any vulnerability issues.

`issues: write` and `pull-requests: write` is the access rights recomended in the documentation (see examples). I only separate issues and pulls to get each one separated at the workflow summary job


> To make the code simpler you could use [`probot/stale`](https://github.com/probot/stale), the bot could serve in other occasions (see [probot/reminders](https://github.com/probot/reminders), [probot.github.io/apps](https://probot.github.io/apps/)...).

I'm starting in actions world and I see that I develop in a rudimentary way :rofl:. I did not know that there was an app with plugins to automate everything in a joint way :face_in_clouds:

Moreover Probot don't have execution summaries like I develop as final workflow step so is dificult to test. :-1:


> Can you test this workflow? Maybe temporarily reducing the days to 1/2.

Here a manual execution: https://github.com/davorpa/free-programming-books/actions/runs/2977616004. Is partial because I fake the job outputs in the workflow in order to test the final job composing the GitHub Summary (not available with Probot as I said).

I'll reduce the timing as you suggest and we could get the verdict on Monday

Reply:
> Moreover Probot don't have execution summaries like I develop as final workflow step so is dificult to test


I really like the summary, an alternative could be using the [audit log](https://docs.github.com/en/organizations/keeping-your-organization-secure/managing-security-settings-for-your-organization/reviewing-the-audit-log-for-your-organization), although few would have access to it.

I am fine with the current solution, however, it needs to be evaluated whether a bot already made can be useful to us in other contexts (see https://github.com/EbookFoundation/free-programming-books/issues/7024#issuecomment-1235645346)

Reply:
Monitoring to get test results....
- davorpa/free-programming-books#9
- davorpa/free-programming-books#10
- davorpa/free-programming-books#11
- davorpa/free-programming-books#12
- davorpa/free-programming-books#13
- davorpa/free-programming-books#14

The results... on Monday :rocket:

Reply:
Fresh results....

https://github.com/davorpa/free-programming-books/actions/runs/2996504685

Reply:
wait for working results for https://github.com/davorpa/free-programming-books/issues/11

Reply:
> wait for working results for [davorpa#11](https://github.com/davorpa/free-programming-books/issues/11) but now everything seems to be working correctly

https://github.com/davorpa/free-programming-books/issues/11 are exempt to stale because is pinned :wink:

You can check logs in ... https://github.com/davorpa/free-programming-books/runs/8213708447?check_suite_focus=true#step:2:138

Reply:
But it was declared "stale" 22 hours ago

Reply:
> But it was declared "stale" 22 hours ago

Don't worry about it.

It was some test to integrate with #7056 or see if other secrets are revealed when is assigned or has other labels and then crash

Current tests:

- https://github.com/davorpa/free-programming-books/issues/11
- https://github.com/davorpa/free-programming-books/issues/16
- https://github.com/davorpa/free-programming-books/issues/18
- https://github.com/davorpa/free-programming-books/issues/19
- https://github.com/davorpa/free-programming-books/issues/20

Reply:
All staled and closed now

Reply:
Nice, do the `exempt-issue-labels` parameters work?

Reply:
> Nice, do the `exempt-issue-labels` parameters work?

Yes of course

Reply:
Being an automatic workflow I don't think there should be in the check section, or am I wrong?